src/ape/api/providers.py

```diff
@@ -308,100 +294,64 @@ def await_confirmations(self) -> "ReceiptAPI":
         return self
 
 
-@abstractdataclass
-class BlockGasAPI:
+class BlockGasAPI(AbstractBaseModel):
     """
     An abstract class for representing gas data for a block.
     """
 
-    gas_limit: int
-    gas_used: int
-    base_fee: Optional[int] = None
-
-    @classmethod
-    @abstractmethod
-    def decode(cls, data: Dict) -> "BlockGasAPI":
-        """
-        Decode data to a :class:`~ape.api.BlockGasAPI`.
+    gas_limit: int = Field(alias="gasLimit")
+    gas_used: int = Field(alias="gasUsed")
+    base_fee: Optional[int] = Field(None, alias="baseFeePerGas")
 
-        Args:
-            data (dict): A dictionary of block-gas properties.
 
-        Returns:
-            :class:`~ape.api.BlockGasAPI`
-        """
-
-
-@abstractdataclass
-class BlockConsensusAPI:
+class BlockConsensusAPI(AbstractBaseModel):
     """
     An abstract class representing block consensus-data,
     such as PoW-related information regarding the block.
     `EIP-3675 <https://eips.ethereum.org/EIPS/eip-3675>`__.
     """
 
     difficulty: Optional[int] = None
-    total_difficulty: Optional[int] = None
+    total_difficulty: Optional[int] = Field(None, alias="totalDifficulty")
 
-    @classmethod
-    @abstractmethod
-    def decode(cls, data: Dict) -> "BlockConsensusAPI":
-        """
-        Decode data to a :class:`~ape.api.BlockConsensusAPI`.
-
-        Args:
-            data (dict): A dictionary of data to decode.
-
-        Returns:
-            :class:`~ape.api.BlockConsensusAPI`
-        """
 
-
-@abstractdataclass
-class BlockAPI:
+class BlockAPI(AbstractBaseModel):
     """
     An abstract class representing a block and its attributes.
     """
 
     gas_data: BlockGasAPI
     consensus_data: BlockConsensusAPI
-    hash: HexBytes
+    hash: Any
     number: int
-    parent_hash: HexBytes
+    parent_hash: Any
     size: int
     timestamp: int
 
-    @classmethod
-    @abstractmethod
-    def decode(cls, data: Dict) -> "BlockAPI":
+    @validator("hash", "parent_hash", pre=True)
+    def validate_hexbytes(cls, value):
         """
-        Decode data to a :class:`~ape.api.BlockAPI`.
-
-        Args:
-            data (dict): A dictionary of data to decode.
-
-        Returns:
-            :class:`~ape.api.BlockAPI`
+        Custom validation method for HexBytes because pydantic treats these
```

Comment:
Do we want doc strs for the validation methods or do just want comments?
Doc strs here will cause the user the maybe think this a method they can call.

I think we should exclude all validation methods from the documentation. You can do this by appending the method name to the `:exclude-members: __repr__, __weakref__, __metaclass__` list in the doc files, like `docs/methoddocs/api.md`

Reply:
Fixing this to an internal `NOTE` for now, but it'll change if we change the data type to something pydantic can validate